addons/binding/org.openhab.binding.geofence/README.md

```diff
@@ -0,0 +1,160 @@
+* **Additional region definitions** - This configuration option is for expert users to create additional internal regions in JSON format. The binding expects an array of regions in the following format:
+
+ ```
+    {
+        "name": "ExtraRegion1",
+        "location": "xx.xxxx,yy.yyyy",
+        "radius": 100,
+        "triggerEvent": true
+    },
+    {
+        "name": "ExtraRegion2",
+        "location": "xx.xxxx,yy.yyyy",
+        "radius": 100,
+        "triggerEvent": false
+    },
+    ...
+ ]
+ ```
+
+### OwnTracks
+
+Install [OwnTracks for Android](https://play.google.com/store/apps/details?id=org.openhab.habdroid) or [OwnTracks for iOS](https://itunes.apple.com/us/app/owntracks/id692424691) on your device.
+
+Go to Preferences/Connection and set:
+
+* **Mode** - select Private HTTP
+* **Host** - **https://<your.ip.address>/geofence/owntracks**
+* **Identification**
+  * Turn Authentication ON
+  * Set username and password to be able to reach your openHAB server
+  * Device ID is not important. Set it to e.g. phone
+  * Tracker ID - This id will identify the tracker as a thing. I use initials here.
+  
+![Image](./docs/owntracks_setup.png)
+
+### GPSLogger
+
+Install [GPSLogger for Android](https://play.google.com/store/apps/details?id=com.mendhak.gpslogger) on your device. After the launch, go to General Options. Enable Start on bootup and Start on app launch.
+
+![Image](./docs/gpslogger_1.png)
+
+Go to Logging details and disable Log to GPX, Log to KML and Log to NMEA. Enable Log to custom URL.
+
+![Image](./docs/gpslogger_2.png)
+
+Right after enabling, the app will take you to the Log to custom URL settings and modify:
+
+* **URL** - Set URL to **https://<username>:<password>@<your.ip.address>/geofence/gpslogger** by replacing <???> with values matching your setup. It’s HIGHLY recommended to use SSL/TLS.
+* **HTTP Body** - Type in the following JSON:
+
+```
+{
+    "_type":"location",
+    "lat":%LAT,
+    "lon":%LON,
+    "tid":"XY",
+    "acc":%ACC,
+    "batt":%BATT,
+    "tst":%TIMESTAMP,
+    "alt":%ALT,
+    "cog":%DIR,
+    "vel":%SPD
+}
+```
+
+**Note**: The value of "tid" is the tracker identified that will identify the tracker as a thing in openHAB. This must be unique for each tracker connected to the same openHAB instance (e.g. family members).
+
+* **HTTP Method** - type in: **POST**
+* **HTTP Headers** - type in: **Content-Type: application/json**
+
+![Image](./docs/gpslogger_3.png)
+
+### Things
+
+It is possible to define things manually or to use the discovery feature of the openHAB. An important detail for both methods is that a tracker is identified by a **tracker id** configured on mobile devices. Make sure these tracker ids are unique in group of trackers connected to a single openHAB instance.
+
```

Comment:
Please provide a table of the config parameters of the thing (and also mention the thing type id here).

Reply:
There are no config parameters from tracker things. The only read only `externalRegionsJSON` parameter stores the external regions read from transition type GPS log records. Actually this can be stored in a  thing property as well, but I think it is useful for debug reasons.

Reply:
There is, see https://github.com/openhab/openhab2-addons/pull/3839#discussion_r215275459.

But wrt `externalRegionsJSON`: This does not make much sense as a configuration parameter nor as a thing property as this information is lost, if the Thing is defined textually and the system is restarted - so if this data is needed for proper region support, it should be either configurable or possible to simply provide region channels without any additional JSON information (probably the better option).

Reply:
What is actually required here? Afaics, the only valuable information is the set of region names (i.e. a `Set<String>`). What is this information used for? To dynamically construct and add the according channels to the Thing? If so, there's no need to persist this as a configuration as the added channels are already persisted (if the Thing is stored in the database).

Reply:
The binding needs a FULL list of regions (including internals and externals) to maintain dynamic channels for trackers. To have this for externals I maintained this JSON list each time the tracker sent a record.

> To dynamically construct and add the according channels to the Thing?

...and to remove if the tracker configuration has changed. This is the reason that persisted channels are not enough.

Reply:
>  I maintained this JSON list

Why JSON and not simply a String list?

> each time the tracker sent a record.

When, how often and what do the trackers send? Always a complete list of regions, or always only one? How do you know if one is removed?

> if the tracker configuration has changed.

You were telling me that trackers do not have any configuration. So what can change about them?